### PR TITLE
Update script to enable concurrency

### PR DESCRIPTION
**Pull request by liliwei25**
Hi! I updated the `c4200m_get_target_sentences.py` script to load multiple corrupt edits at once so that we can check multiple shards within one loop. I also increased the batch size when loading the c4 dataset and this improved the performance drastically! Please feel free to comment and let me know if this script is helpful or if any edits need to be done. Thank you!

### CODE REVIEW

**Comment by liliwei25** [comment]
I've signed the CLA. Please help to rescan. Thank you!

File: c4200m_get_target_sentences.py [hunk 1]
```diff
@@ -5,44 +5,75 @@
...
+import re
 
 LOGGING_STEPS = 100000
+BATCH_SIZE = 64
 
+def rreplace(string, old, new):
+  return new.join(string.rsplit(old, 1))
+
+def get_file_paths(edits_tsv_path, concurrent_runs):
+  start_index_string = re.search('(\d+)', edits_tsv_path).group(0)
```
Comment (fstahlberg) [inline c4200m_get_target_sentences.py]: This could go wrong if the edit_tsv_path contains a number somewhere else.

File: c4200m_get_target_sentences.py [hunk 2]
```diff
@@ -5,44 +5,75 @@
 
 from absl import app
 import tensorflow_datasets as tfds
+import multiprocessing as mp
+import re
 
 LOGGING_STEPS = 100000
+BATCH_SIZE = 64
 
+def rreplace(string, old, new):
```
Comment (fstahlberg) [inline c4200m_get_target_sentences.py]: Slightly hard to parse. Please add a short docstring describing what this function is doing.

File: c4200m_get_target_sentences.py [hunk 3]
```diff
@@ -5,44 +5,75 @@
...
+  
+  edits_tsv_paths = get_file_paths(edits_tsv_path, concurrent_runs)
+  output_tsv_paths = get_file_paths(output_tsv_path, concurrent_runs)
 
-  print("Loading C4_200M target sentence hashes from %r..." % edits_tsv_path)
-  remaining_hashes = set()
-  with open(edits_tsv_path) as edits_tsv_reader:
-    for tsv_line in edits_tsv_reader:
-      remaining_hashes.add(tsv_line.split("\t", 1)[0])
+  remaining_hashes = dict()
```
Comment (fstahlberg) [inline c4200m_get_target_sentences.py]: why changing it from a set to a dict?

**Review by fstahlberg — CHANGES_REQUESTED** [review]
Thanks for your PR - that will certainly help to speed things up. However, even if that results in some duplicate code, could you add a separate script for parallel processing instead of changing the old one? The simple reason is that I'd prefer to keep the changes to the original script to a minimum to make sure that the dataset in it's initial form is still reproducible as closely as possible.

File: c4200m_get_target_sentences.py (hunk 3 above)
Comment (liliwei25) [reply c4200m_get_target_sentences.py]: since we are going through multiple files, we need to store the index of the original file in the list. i.e. `"hash": 0`

File: c4200m_get_target_sentences.py (hunk 1 above)
Comment (liliwei25) [reply c4200m_get_target_sentences.py]: thanks for the tip! let me fix this!

**Comment by fstahlberg** [comment]
Looking good, thanks!

**Comment by liliwei25** [comment]
thanks @fstahlberg! 😄
